### PR TITLE
Migrate to the Objective-C interchange format

### PR DESCRIPTION
**Pull request by jverkoey**
This change introduces new APIs that are compatible with the new interchange format introduced in MotionInterchange [v1.5.0](https://github.com/material-motion/motion-interchange-objc/releases/tag/v1.5.0). The API arguments have also been reordered to be somewhat more intuitive.

Note that this is not intended to be a breaking change.

### CODE REVIEW

File: src/MDMMotionAnimator.h [hunk 1]
```diff
@@ -69,73 +62,91 @@ NS_SWIFT_NAME(MotionAnimator)
...
+               completion:(nullable void(^)(BOOL didComplete))completion;
+
+/**
+ If enabled, explicitly-provided values will be reversed before animating.
+
+ This property only affects the animateWithTraits:between:... family of methods.
+
+ Disabled by default.
+ */
+@property(nonatomic, assign) BOOL shouldReverseValues;
```
Comment (romoore) [inline src/MDMMotionAnimator.h:117]: If these are supposed to be "grouped", consider bundling them into a category.

File: src/private/CAMediaTimingFunction+MotionAnimator.h [hunk 2]
```diff
@@ -18,6 +18,6 @@
 #import <CoreGraphics/CoreGraphics.h>
 #import <QuartzCore/QuartzCore.h>
 
-// Returns a timing function with the given control points.
+// Returns a traits function with the given control points.
 FOUNDATION_EXPORT
 CAMediaTimingFunction* MDMTimingFunctionWithControlPoints(CGFloat controlPoints[4]);
```
Comment (romoore) [inline src/private/CAMediaTimingFunction+MotionAnimator.h]: Typo? This looks like a substitution error.

File: src/private/CABasicAnimation+MotionAnimator.h [hunk 3]
```diff
@@ -20,9 +20,9 @@
 #import <CoreGraphics/CoreGraphics.h>
 #import <QuartzCore/QuartzCore.h>
 
-// Returns a basic animation configured with the provided timing and scale factor.
+// Returns a basic animation configured with the provided traits and scale factor.
 FOUNDATION_EXPORT
-CABasicAnimation *MDMAnimationFromTiming(MDMMotionTiming timing, CGFloat timeScaleFactor);
```
Comment (romoore) [inline src/private/CABasicAnimation+MotionAnimator.h:25]: Should this function be renamed and announced as becoming deprecated?

File: src/private/MDMUIKitValueCoercion.h [hunk 4]
```diff
@@ -16,7 +16,7 @@
 
 #import <Foundation/Foundation.h>
 
-// Coerces the following UIKit values to Core Animation values:
```
Comment (romoore) [inline src/private/MDMUIKitValueCoercion.h:19]: Typo?

File: tests/unit/HeadlessLayerImplicitAnimationTests.swift [hunk 5]
```diff
@@ -97,7 +97,7 @@ class HeadlessLayerImplicitAnimationTests: XCTestCase {
 
   // Verifies the somewhat counter-intuitive fact that CATransaction's animation duration always
   // takes precedence over UIView's animation duration. This means that animating a headless layer
-  // using UIView animation APIs may not result in the expected timings.
+  // using UIView animation APIs may not result in the expected traitss.
```
Comment (romoore) [inline tests/unit/HeadlessLayerImplicitAnimationTests.swift]: Typo

File: tests/unit/InitialVelocityTests.swift [hunk 6]
```diff
@@ -51,7 +51,7 @@ class InitialVelocityTests: XCTestCase {
     addedAnimations.flatMap { $0 as? CASpringAnimation }.forEach { animation in
       XCTAssertEqual(animation.initialVelocity, 0.5,
                      "from: \(animation.fromValue!), "
-                      + "to: \(animation.toValue!), "
+                      + "layer: \(animation.toValue!), "
```
Comment (romoore) [inline tests/unit/InitialVelocityTests.swift]: Typo? Here and below

**Review by romoore — COMMENTED** [review]
Add tests for the soon-to-be-deprecated behavior to confirm no breaking changes?

File: src/MDMMotionAnimator.h (hunk 1 above)
Comment (jverkoey) [reply src/MDMMotionAnimator.h:117]: This is the canonical way to have APIs grouped in API docs. See the attached jazzy output:

screen shot 2017-12-13 at 3 35 39 pm

File: src/private/CABasicAnimation+MotionAnimator.h (hunk 3 above)
Comment (jverkoey) [reply src/private/CABasicAnimation+MotionAnimator.h:25]: Done.

File: src/private/CAMediaTimingFunction+MotionAnimator.h (hunk 2 above)
Comment (jverkoey) [reply src/private/CAMediaTimingFunction+MotionAnimator.h]: Done.

File: src/private/MDMUIKitValueCoercion.h (hunk 4 above)
Comment (jverkoey) [reply src/private/MDMUIKitValueCoercion.h:19]: Done.

File: tests/unit/HeadlessLayerImplicitAnimationTests.swift (hunk 5 above)
Comment (jverkoey) [reply tests/unit/HeadlessLayerImplicitAnimationTests.swift]: Done.

File: tests/unit/InitialVelocityTests.swift (hunk 6 above)
Comment (jverkoey) [reply tests/unit/InitialVelocityTests.swift]: Done.

File: src/MDMMotionAnimator.h (hunk 1 above)
Comment (romoore) [reply src/MDMMotionAnimator.h:117]: Ack.

**Comment by jverkoey** [comment]
Can haz review?

File: examples/CalendarCardExpansionExample.m [hunk 7]
```diff
@@ -40,82 +40,68 @@ @implementation CalendarCardExpansionExampleViewController {
...
-                      toLayer:_chipView.layer
-                   withValues:@[ @(CGRectGetMidY(chipFrame)), @(CGRectGetMidY(headerFrame)) ]
-                      keyPath:MDMKeyPathY];
-  [animator animateWithTiming:timing.chipHeight
-                      toLayer:_chipView.layer
-                   withValues:@[ @([self chipCornerRadius]), @0 ]
-                      keyPath:MDMKeyPathCornerRadius];
+  [animator animateWithTraits:traits.chipHeight between:@[ @(chipFrame.size.height),
+                                                           @(headerFrame.size.height) ]
+                        layer:_chipView.layer keyPath:MDMKeyPathHeight];
```
Comment (samnm) [inline examples/CalendarCardExpansionExample.m]: Is it not preferred to keep one argument per line here?

Comment (jverkoey) [reply examples/CalendarCardExpansionExample.m]: Done.

**Review by samnm — APPROVED** [review]

**Pull request merged**